python/tank/bootstrap/cached_configuration.py

```diff
@@ -239,6 +251,9 @@ def update_configuration(self):
                     os.path.join(self._path.current_os, "install", "core")
                 )
                 log.debug("Previous core restore complete...")
+        else:
+            self._delete_old_backups(os.path.dirname(os.path.dirname(config_backup_path)))
```

Comment:
I am paranoid about deleting stuff. The idea of introducing a blanket delete feels really unsafe to me. What if some client have symlinked something here? what does `../..` mean in that case? it's not well defined. With toolkit, we have to constantly assume that people do lost of weird and wonderful types of customizations.

I suggest that rather than trying to delete *all* backups, we just delete the things we have moved further up in the code. This seems safer to me.

So that would give us the following flow:

1. we detect that a new config is needed so we need to make way for it
2. we move config and core out of the config install location into x and y
3. we now install the new config and core
4. if something goes wrong during 3) we move x and y back into their previous location, report errors and continue.
5. on the other hand, if the update succeeded, that's great - all good and we no longer need x and y so we delete it.

@gomezll how does this seem? Unless i am missing something this is the approach i would suggest - we only delete stuff we create. That approach means we won't cause any issues if a client has some strange customizations etc in place.